content/v2.0/actions/parameters.md

```diff
@@ -0,0 +1,308 @@
+---
+title: Parameters
+order: 30
+---
+
+Parameters are taken from the Rack env, wrapped into the `Params` object, and passed as an argument to `#handle` method.
+
+They are similar to a Ruby `Hash`, but they offer an expanded set of features.
+
+## Sources
+
+Params can come from:
+
+  * [Router variables](/v2.0/router/basic-usage) (eg. `/books/:id`)
+  * Query string (eg. `/books?title=Hanami`)
+  * Request body (eg. a `POST` request to `/books`)
+
+## Access
+
+To access the value of a param, we can use the _subscriber operator_ `#[]`.
+
+```ruby
+# app/controllers/dashboard/index.rb
+
+module Bookshelf
+  module Actions
+    module Books
+      class Index < Action
+        def handle(req, res)
+          req.body = "Query string: #{ req.params[:q] }"
+        end
+      end
+    end
+  end
+end
+```
+
+If we visit `/dashboard?q=foo`, we should see `Query string: foo`.
+
+### Symbol Access
+
+Params and nested params can be referenced **only** via symbols.
+
+```ruby
+params[:q]
+params[:book][:title]
+```
+
+Now, what happens if the parameter `:book` is missing from the request?
+Because `params[:book]` is `nil`, we can't access `:title`.
+In this case Ruby will raise a `NoMethodError`.
+
+We have a safe solution for our problem: `#get`.
+It accepts a list of symbols, where each symbol represents a level in our nested structure.
+
+```ruby
+params.get(:book, :title)             # => "Hanami"
+params.get(:unknown, :nested, :param) # => nil instead of NoMethodError
+```
```

Comment:
Oh noez, I had no idea this exists. Too late now but...maybe not? If we also provide `dig` then please do change it to `dig` here, if not...then oh well, something we can address in 3.0.

Reply:
Both work the same - replaced with `dig`. We can remove the `get` if no needed in the upcoming releases.